File: velocity/src/main/java/ru/privatenull/pnauth/velocity/VelocityCommandSource.java
package ru.privatenull.pnauth.velocity;

import com.velocitypowered.api.command.CommandSource;
import com.velocitypowered.api.proxy.Player;

import java.util.UUID;

final class VelocityCommandSource implements ru.privatenull.pnauth.command.CommandSource {
    private final CommandSource source;

    VelocityCommandSource(CommandSource source) {
        this.source = source;
    }

    @Override
    public UUID uniqueId() {
        return source instanceof Player player ? player.getUniqueId() : null;
    }

    @Override
    public String username() {
        return source instanceof Player player ? player.getUsername() : null;
    }

    @Override
    public boolean isPlayer() {
        return source instanceof Player;
    }

    @Override
    public boolean hasPermission(String permission) {
        return source.hasPermission(permission);
    }

    CommandSource source() {
        return source;
    }
}

File: velocity/src/main/kotlin/ru/privatenull/pnauth/velocity/VelocityCommandSource.kt
package ru.privatenull.pnauth.velocity

import com.velocitypowered.api.command.CommandSource
import com.velocitypowered.api.proxy.Player
import ru.privatenull.pnauth.command.CommandSource as PnCommandSource
import java.util.UUID

internal class VelocityCommandSource(
    private val source: CommandSource
) : PnCommandSource {

    override fun uniqueId(): UUID? {
        return (source as? Player)?.uniqueId
    }

    override fun username(): String? {
        return (source as? Player)?.username
    }

    override fun isPlayer(): Boolean {
        return source is Player
    }

    override fun hasPermission(permission: String): Boolean {
        return source.hasPermission(permission)
    }

    fun source(): CommandSource = source
}
